knowledge-base/actions/benc-uk/action-security.yml

```diff
@@ -0,0 +1,10 @@
+name: 'Workflow Dispatch'
+github-token:
+  action-input:
+    input: token
+    is-default: false
+  permissions:
+    actions: read
```

Comment:
@vandana41  if you [check_this](https://github.com/benc-uk/workflow-dispatch/blob/827565b908f387ffd483c84312273ae185c06c8a/src/main.ts#L51), a post request is being sent to dispatch a workflow. As `dispatch_workflow` comes under `actions` [check_here](https://docs.github.com/en/rest/actions/workflows#create-a-workflow-dispatch-event), so the permission would be `actions: write`. And the reason would be `actions-reason: to list and dispatch workflows`